### PR TITLE
PMM-6457 Add API to manage PXC via dbaas controller.

### PR DESCRIPTION
**Pull request by askomorokhov**
[PMM-6457](https://jira.percona.com/browse/PMM-6457)

---

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1124

- [x] Links to other linked pull requests (optional).
- https://github.com/Percona-Lab/pmm-api-tests/pull/125
- https://github.com/percona-platform/dbaas-controller/pull/58
- https://github.com/percona/pmm-managed/pull/468
- https://github.com/percona/pmm/pull/582
---
- [x] Tests passed.
- [x] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [x] Update API dependency.
- [x] Update jira ticket description if needed.
- [x] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -595,6 +598,13 @@ func main() {
 
 	authServer := grafana.NewAuthServer(grafanaClient, awsInstanceChecker)
 
+	// TODO:
+	dbaasControllerConn, err := grpc.Dial("127.0.0.1:20201", grpc.WithInsecure())
```
Comment (BupycHuk) [inline main.go:602]: maybe it's better to extract it to separate function like we did for qanClient

File: Gopkg.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ required = [
 
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  branch = "PMM-6457-api-db-cluster"
```
Comment (BupycHuk) [inline Gopkg.toml:19]: PR for this branch is still draft.

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Can we write tests which will be skipped by CI, but can be run locally?

File: main.go (hunk 1 above)
Comment (askomorokhov) [reply main.go:602]: Yes, that why TODO here.

**Comment by AlekSi** [comment]
Please also resolve merge conflicts

**Review by artemgavrilov — APPROVED** [review]

File: main.go [hunk 3]
```diff
@@ -114,12 +114,13 @@ func addLogsHandler(mux *http.ServeMux, logs *supervisord.Logs) {
...
-	agentsRegistry *agents.Registry
-	grafanaClient  *grafana.Client
-	checksService  *checks.Service
+	db                  *reform.DB
+	prometheus          *prometheus.Service
+	server              *server.Server
+	agentsRegistry      *agents.Registry
+	grafanaClient       *grafana.Client
+	checksService       *checks.Service
+	dbaasControllerConn *dbaas.Client
```
Comment (percona-csalguero) [inline main.go:123]: Just a comment: I would name is dbassClient and not ControllerConn. 
Only to make it more clear

File: services/management/dbaas/xtradb_cluster_service.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	in := dbaascontrollerv1beta1.UpdateXtraDBClusterRequest{
+		KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
+			Kubeconfig: kubernetesCluster.KubeConfig,
+		},
+		Name: req.Name,
```
Comment (percona-csalguero) [inline services/management/dbaas/xtradb_cluster_service.go:139]: At line 130 we are getting the cluster by name and here we are re-setting the name but i makes me think?
1) Is it possible to rename a cluster?
2) What happens is `req.Name` is a duplicated name?

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: services/management/dbaas/xtradb_cluster_service.go (hunk 4 above)
Comment (askomorokhov) [reply services/management/dbaas/xtradb_cluster_service.go:139]: As you can see `UpdateXtraDBCluster` is not implemented yet - so it is too early fo fix here anything.
https://github.com/percona-platform/dbaas-controller/blob/main/service/cluster/xtra_db_cluster.go#L128-L131

File: main.go (hunk 3 above)
Comment (askomorokhov) [reply main.go:123]: Well, I agree that is it not the best name.
But it is still a connection to the *Controller* of DBaaS, but not directly to DBaaS.
So, I hesitate here about renaming.

**Review by percona-csalguero — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: main.go (hunk 3 above)
Comment (BupycHuk) [reply main.go:123]: But struct is dbaas.Client, right?

Comment (BupycHuk) [reply main.go:123]: dbaasControllerClient?

File: services/management/dbaas/xtradb_cluster_service_test.go [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+const kubeconfTest = `
+	{
+		"apiVersion": "v1",
+		"kind": "Config",
+		"users": [
+			{
+				"name": "percona-xtradb-cluster-operator",
+				"user": {
+					"token": "some-tocken"
```
Comment (BupycHuk) [inline services/management/dbaas/xtradb_cluster_service_test.go:49]:
```suggestion
					"token": "some-token"
```

**Review by BupycHuk — APPROVED** [review]
Looks good to me, just a few comments related to naming.

**Review by AlekSi — APPROVED** [review]

File: services/management/dbaas/xtradb_cluster_api_client.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+	return &Client{
+		XtraDBClusterAPIClient: controllerv1beta1.NewXtraDBClusterAPIClient(conn),
+	}, nil
+}
+
+func dial(ctx context.Context, address string) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{
+		grpc.WithUserAgent("pmm-managed/" + version.Version),
+		grpc.WithInsecure(),
```
[the remark this answers is not in the document]
Comment (askomorokhov) [reply services/management/dbaas/xtradb_cluster_api_client.go:54]: Changed

**Comment by AlekSi** [comment]
There are conflicts now

**Pull request merged**